rundeck_action/action_types/rundeck.ts

```diff
@@ -0,0 +1,259 @@
+export function getActionType({ logger }: { logger: Logger }): ActionType {
+  return {
+    id: '.rundeck',
+    name: 'rundeck',
+    validate: {
+      config: schema.object(configSchemaProps),
+      secrets: SecretsSchema,
+      params: ParamsSchema,
+    },
+    executor: curry(executor)({ logger }),
+  };
+}
+
+// action executor
+// * Need API Key for calling PD Rest APIs
+export async function executor(
+  { logger }: { logger: Logger },
+  execOptions: ActionTypeExecutorOptions
+): Promise<ActionTypeExecutorResult> {
+  const actionId = execOptions.actionId;
+  const { method, rundeckBaseUrl, rundeckApiVersion, jobId, headers = {} } = execOptions.config as ActionTypeConfigType;
+  const { rundeckApiToken } = execOptions.secrets as ActionTypeSecretsType;
+  const { dedupKey, jobParams } = execOptions.params as ActionParamsType;
+
+  Object.assign(headers, {"X-Rundeck-Auth-Token": rundeckApiToken})
+
+  const rundeckBaseUrlWithoutSlash = rundeckBaseUrl.endsWith('/') ? rundeckBaseUrl.slice(0, -1) : rundeckBaseUrl;
+
+  const rundeckApiUrl = `${rundeckBaseUrlWithoutSlash}/api/${rundeckApiVersion}/job/${jobId}/executions`;
+
+  const rundeckResult: Result<AxiosResponse, AxiosError> = await promiseResult(
+    axios.request({
+      method,
+      url: rundeckApiUrl,
+      headers,
+      data: jobParams
+    })
+  ); 
+
+  if (isOk(rundeckResult)) {
+    const {
+      value: { status, statusText, data: { permalink: executionLink }},
+    } = rundeckResult;
+
+    let {
+      value: {data: responseData}
+    } = rundeckResult
+
+    // pageduty APIs
+    const pdIncidentListApiBaseUrl = 'https://api.pagerduty.com/incidents?incident_key=';
+    const pdAddNoteApiBaseUrl = 'https://api.pagerduty.com/incidents/';
+
+    if (dedupKey) {
+
+      const { headers = {} } = execOptions.config as ActionTypeConfigType;
+      const { pdApiKey } = execOptions.secrets as ActionTypeSecretsType;
+
+      Object.assign(headers, {"authorization": `Token token=${pdApiKey}`})
+
+      const pdIncidentListApiOptions: PagerDutyApiOptions = {
+        actionId,
+        url: `${pdIncidentListApiBaseUrl}${dedupKey}`,
+        method: "get",
+        headers,
+        dedupKey,
+        data: {},
+        logger,
+      }
+
+      const pdIncidentListResult: PagerDutyApiResult = await callPagerDutyApi(pdIncidentListApiOptions);
+
+      if (pdIncidentListResult.status === "error") {
+        logger.warn(`error on ${actionId} rundeck action: ${pdIncidentListResult.message}`);
+        return errorPagerDutyProcess(actionId, pdIncidentListResult.message);
+      }
+
+      // incident list is empty
+      if (pdIncidentListResult.data.incidents.length == 0) {
+        const message: string = `Pager Duty incident list requested by dedupKey(incident_key), "${dedupKey}", is empty.`
+        return errorPagerDutyProcess(actionId, message);
+      }
+
+      // call PD create note api
+      const { id: incidentId } = pdIncidentListResult.data.incidents.slice(-1)[0]
```

Comment:
```suggestion
      const incidentId = pdIncidentListResult.data.incidents.pop().id
```

Questionable readability in the object destructoring for a single value.